src/vm.ts

```diff
@@ -237,9 +240,12 @@ export class CosmWasmVM {
   }
 
   protected do_addr_canonicalize(source: Region, destination: Region): Region {
-    throw new Error('not implemented');
+    let canonical = this.bech32.fromWords(
+      this.bech32.decode(source.str).words
+    );
+    destination = this.allocate_bytes(Buffer.from(canonical));
+    return new Region(this.exports.memory, 0);
   }
```

Comment:
actually, this is a dirty fix, to `always` return a 0 ptr to the VM, so it does always pass - normally, if there is some error from the bech32 decoding lib, we should allocate some Region pointer containing that error string and return that then. CosmWasmVM will fire an error with the string at that ptr, if the Region ptr is !== 0

Reply:
Kapan bulan?